kuma/wiki/models.py

```diff
@@ -811,6 +809,10 @@ def revert(self, revision, user, comment=None):
         revision.created = datetime.now()
         revision.creator = user
         revision.save()
+        # TODO: change to primary key check instead of object comparison
+        if revision.document.original == self:
+            revision.based_on = revision
+        revision.save()
```

Comment:
I would also move this inside the `if` because when this is not True we'd be calling save 2 times right after the other with no changes. Unless I'm reading this wrong.